Sources/SwiftFormat/Utilities/FileIterator.swift

```diff
@@ -169,3 +179,23 @@ private func fileType(at url: URL) -> FileAttributeType? {
   // Linux.
   return try? FileManager.default.attributesOfItem(atPath: url.path)[.type] as? FileAttributeType
 }
+
+/// Returns true if the file should be processed.
+/// Directories are always processed.
+/// Other files are processed if there is not a 
+/// ignore file in the containing directory or any of its parents.
+private func inputShouldBeProcessed(at url: URL) -> Bool {
+  guard fileType(at: url) != .typeDirectory else {
+    return true
+  }
+      
+  var containingDirectory = url.absoluteURL.standardized
+  repeat {
+    containingDirectory.deleteLastPathComponent()
+    let candidateFile = containingDirectory.appendingPathComponent(FileIterator.ignoreFileName)
+    if FileManager.default.isReadableFile(atPath: candidateFile.path) {
```

Comment:
More than just checking for the presence of the file, I think we go the extra step of validating that it only contains `*` so that users aren't surprised later when we add more functionality and it subtly changes the behavior based on those contents. If it contains anything else, we should throw an error saying it's the only supported form (and turn that into a nice human-readable error message when we exit).

Reply:
If we want to upgrade to gitignore like sytanx in the future, should we make the default value `**/*`?

Reply:
AFAIK, `*` is sufficient because according to the [documentation](https://git-scm.com/docs/gitignore):

* "An asterisk '`*`' matches anything except a slash."
* "If there is a separator at the beginning or middle (or both) of the pattern, [...]. Otherwise the pattern may also match at any level below the `.gitignore` level."
* "If there is a separator at the end of the pattern then the pattern will only match directories, otherwise the pattern can match both files and directories."

Reply:
Sorry, got distracted, but am looking at this now.

One complication is that FileIterator currently doesn't throw, and instead relies on the caller to generate diagnostics (eg for non-existent paths). 

In order to fully validate the ignore file inside FileIterator we'd either need to be able to throw, or be able to emit diagnostics, or we'd need to spot an invalid ignore file and return its path (or a special pseudo-path), then spot that path in the front end and emit the diagnostic there. All of which is doable but a bit ugly.